R/pb_download.R

```diff
@@ -1,29 +1,38 @@
 #' Download data from an existing release
 #'
-#' @param file name or vector of names of files to be downloaded. If `NULL`,
+#' @param file character: vector of names of files to be downloaded. If `NULL`,
 #' all assets attached to the release will be downloaded.
-#' @param dest name of vector of names of where file should be downloaded.
-#' Can be a directory or a list of filenames the same length as `file`
-#' vector. Any directories in the path provided must already exist.
-#' @param overwrite Should any local files of the same name be overwritten?
-#'  default `TRUE`.
-#' @param ignore a list of files to ignore (if downloading "all" because
-#'  `file=NULL`).
+#' @param dest character: path to destination directory (if length one) or
+#' vector of destination filepaths the same length as `file`.
+#' Any directories in the path provided must already exist.
+#' @param overwrite boolean: should any local files of the same name be overwritten? default `TRUE`
+#' @param ignore character: vector of files to ignore (used if downloading "all" via `file=NULL`)
 #' @inheritParams pb_upload
 #'
 #' @export
-#' @examples \dontrun{
-#'  ## Download a specific file.
-#'  ## (dest can be omitted when run inside and R project)
-#'  piggyback::pb_download("iris.tsv.gz",
-#'                         repo = "cboettig/piggyback-tests",
-#'                         dest = tempdir())
-#' }
-#' \dontrun{
-#'  ## Download all files
-#'  piggyback::pb_download(repo = "cboettig/piggyback-tests",
-#'                         dest = tempdir())
-#'
+#' @examples \donttest{
```

Comment:
I believe CRAN decides to run `\donttest` occasionally (effectively donttest is acceptable for tests that would otherwise run long enough to trigger a NOTE about a long test).  If CRAN runs this and the network connection fails (as it often does on their overloaded network), we get a nastygram about needing to remind us that tests must fail "gracefully" when accessing internet.  

It looks like you avoid the problem with `try(`, but I think that still displays in the docs and thus looks a bit confusing to users?  I have adopted the `@examplesIf()` notation in roxygen, which I think is a bit cleaner.  (e.g. like so https://github.com/apache/arrow/blob/7c3480e2f028f5881242f227f42155cf833efee7/r/R/udf.R#L55 )

Reply:
Does pkgdown provide the NOT_CRAN envvar? That’s my usual goal with these

Reply:
I think the idea is that on CRAN, `NOT_CRAN` is not defined, and so `identical(Sys.getenv("NOT_CRAN"), "true")` evaluates to FALSE, and things don't run on CRAN. 

I think NOT_CRAN is defined as TRUE when testthat runs on GitHub actions or locally.  (I think this is set internally by `rcmdcheck` / `devtools::check()` call, but not set in the official `R CMD CHECK` cli call that CRAN uses).

Reply:
Yeah, I'm familiar with the NOT_CRAN envvar usage in test/check scenarios, I think my goal with using try() in this scenario is so that the example runs on pkgdown and demonstrates successfully downloading the file in the comments, e.g. https://nflreadr.nflverse.com/reference/load_pbp.html#ref-examples shows the print output from `load_pbp()`